packages/x-utils/src/types/utils.types.ts

```diff
@@ -100,3 +100,45 @@ export type PropertyType<
   : SomeObject extends any[]
   ? SomeObject[number]
   : never;
+
```

Comment:
would be better to split this big file in 2 smaller?

Reply:
:sisi3: :callaito:

Reply:
Yes, both have been refactored.